### PR TITLE
Add support mysql unix domain socket

### LINKED ISSUE #228: Support connecting to MySQL via Unix Domain Socket
Currently, IcingaDB only allows connecting to MySQL using a TCP connection. MySQL also supports Unix Domain Sockets.

https://github.com/Icinga/icingadb/blob/2e1610a1cc1333ebad749ed27fa92f50a00343dd/main.go#L110

The MySQL driver in use already supports Unix Domain Sockets using a data source name like this:

```
username:@unix(/run/mysqld/mysqld.sock)/icingadb
```

### PR DESCRIPTION
**Pull request by Yonas-net**
resolves #228

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
I've got a better idea than an additional config option: check whether the given host is an absolute path. If yes, use it as \*nix socket path.

File: main.go [hunk 1]
```diff
@@ -93,8 +95,12 @@ func main() {
...
 	}
 
-	if err := config.ParseConfig(*configPath); err != nil {
-		log.Fatalf("Error reading config: %v", err)
+	if !filepath.IsAbs(*configPath) {
+		if err := config.ParseConfig(*configPath); err != nil{
+			log.Fatalf("Error reading config: %v", err)
+		}
+	} else {
+		config.SetMysqlUnixUser(*dbUser)
```
Comment (Al2Klimov) [inline main.go:103]: Why not to read the config in this case?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (Yonas-net) [reply main.go:103]: I don't really understand what you mean?

Comment (Al2Klimov) [reply main.go:103]: I mean: now if `filepath.IsAbs(*configPath)` you don’t `config.ParseConfig(*configPath)`.

Comment (Yonas-net) [reply main.go:103]: Because ``ParseConfig()`` parses only ``*.ini`` files and not ``*.socket`` files.

Comment (Al2Klimov) [reply main.go:103]: No one ever said anything about not parsing ini files, nor "parsing socket files".

Comment (Al2Klimov) [reply main.go:103]: ... so why you’re doing this?

File: main.go [hunk 2]
```diff
@@ -106,10 +107,14 @@ func main() {
 
 	redisConn := connection.NewRDBWrapper(redisInfo.Host+":"+redisInfo.Port, redisInfo.Password, redisInfo.PoolSize)
 
-	mysqlConn, err := connection.NewDBWrapper(
-		mysqlInfo.User+":"+mysqlInfo.Password+"@tcp("+mysqlInfo.Host+":"+mysqlInfo.Port+")/"+mysqlInfo.Database,
-		mysqlInfo.MaxOpenConns,
-	)
+	var dbDSN string
+	if filepath.IsAbs(mysqlInfo.Host) {
+		dbDSN = mysqlInfo.User+":@unix("+mysqlInfo.Host+")/"+mysqlInfo.Database
```
Comment (Al2Klimov) [inline main.go:112]: And the password?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Comment by Al2Klimov** [comment]
Write a test protocol for both protocols.

**Comment by Yonas-net** [comment]
**Before**: not supported 

**After**:

``cat icingadb.ini`` 

```
[redis]
host="127.0.0.1"
port=6379
;password="icingadb"

[mysql]
host="/tmp/mysql.sock"
user="icingadb"
password="password"

[logging]
level="info"
```

Running icingadb:

```
INFO[0000] Connecting to Redis                          
INFO[0000] Connecting to MySQL                          
INFO[0000] Redis connection established                 
INFO[0000] Starting heartbeat listener                  
INFO[0000] SQL connection established 
```

**Comment by Al2Klimov** [comment]
@N-o-X Not demanding docs until you comment the implementation.

**Comment by N-o-X** [comment]
Works for me and the implementation is also fine.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Document this feature.

File: doc/03-Configuration.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ Data resource where IcingaDB stores synced data and historical events.
 
 Option                   | Description
 -------------------------|-----------------------------------------------
-host                     | **Optional.** MySQL host. Defaults to `127.0.0.1`.
+host                     | **Optional.** MySQL host. Defaults to `127.0.0.1`. It is also possible to use unix socket here by entering the absolute path to ``mysql.sock``.
```
Comment (Al2Klimov) [inline doc/03-Configuration.md:26]: Why double \`?

Comment (Al2Klimov) [inline doc/03-Configuration.md:26]: Combine and shorten that:

"MySQL host." => "MySQL host or absolute Unix socket path."

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Squash your commits.

**Review by N-o-X — APPROVED** [review]

**Pull request merged**